### ISSUE TITLE
Info: MinioS3Error MinioError: The security token included in the request is invalid

### ISSUE THREAD

**Issue by doralines** [xtyxtyx/minio-dart#88]
I think my sessionToken is free of exceptions, and I have verified it in the sdk of the other end

**Comment by doralines** [comment]
What I use is fputObject()
![image](https://github.com/xtyxtyx/minio-dart/assets/150234314/23c9bbc7-0582-4cbd-8810-107458a2a121)

**Comment by doralines** [comment]
What I use is fputObject()
![image](https://github.com/xtyxtyx/minio-dart/assets/150234314/727e96bd-155d-4a6e-9edc-c993b53f617d)

**Comment by serious198706** [comment]
Add a line to `minio_uploader.dart`  `add_stream` function:
```
if (!client.enableSHA256) {
  md5digest = md5.convert(chunk).bytes;
  headers['Content-MD5'] = base64.encode(md5digest);
}
headers['x-amz-security-token'] = minio.sessionToken ?? '';     // add this line
```

**Comment by jpetro416** [comment]
I've added @serious198706 answer to my fork (conditionally) if anyone needs it...


`Future addStream(Stream<Uint8List> stream) async {
  await for (var chunk in stream) {
    List<int>? md5digest;
    final headers = <String, String>{};
    headers.addAll(metadata);
    headers['Content-Length'] = chunk.length.toString();
    if (!client.enableSHA256) {
      md5digest = md5.convert(chunk).bytes;
      headers['Content-MD5'] = base64.encode(md5digest);
    }
    if (minio.sessionToken != null) {
      headers['x-amz-security-token'] = minio.sessionToken;  // add this line
    }
`


  minio:
    git:
      url: https://github.com/jpetro416/minio-dart

**Comment by lijy91** [comment]
Resolved, please update to [`v3.5.1`](https://pub.dev/packages/minio/changelog#351)
